waiter/src/waiter/auth/authentication.clj

```diff
@@ -28,21 +29,30 @@
     "Attaches middleware that enables the application to perform authentication.
      The middleware should
      - either issue a 401 challenge asking the client to authenticate itself,
-     - or upon successful authentication populate the request with :authorization/user and :authenticated-principal"))
+     - or upon successful authentication populate the request with :authorization/user and :authorization/principal"))
 
 (defn- add-cached-auth
   [response password principal]
   (cookie-support/add-encoded-cookie response password AUTH-COOKIE-NAME [principal (System/currentTimeMillis)] 1))
 
+(defn assoc-auth-params
+  "Associate values for authenticated user in the request."
+  ([request principal]
+   (assoc-auth-params request principal (first (str/split principal #"@" 2))))
+  ([request principal user]
+   (assoc request
+          :authorization/principal principal
+          :authorization/user user)))
+
 (defn handle-request-auth
   "Invokes the given request-handler on the given request, adding the necessary
   auth headers on the way in, and the x-waiter-auth cookie on the way out."
-  [request-handler request user principal password]
-  (-> request
-      (assoc :authorization/user user
-             :authenticated-principal principal)
-      request-handler
-      (add-cached-auth password principal)))
+  [handler request user principal password]
+  (let [assoc-auth-params-fn #(assoc-auth-params % principal user)
+        handler' (middleware/wrap-update handler assoc-auth-params-fn)]
+    (-> request
```

Comment:
I find `wrap-context` much more simpler to read and use than `wrap-update`.
This could have been:
```
auth-params-map (auth-params-map principal user)
handler' (middleware/wrap-context handler auth-params-map)
```

where:
```
(defn auth-params-map
  "Returns a map for authenticated user."
  ([principal]
   (auth-params-map principal (first (str/split principal #"@" 2))))
  ([principal user]
   {:authorization/principal principal
    :authorization/user user}))
```